### DISCUSSION TITLE
Difficulty including DAGMC via cmakelists FetchContent

### DISCUSSION THREAD

**Discussion by shimwell** [svalinn/DAGMC#959] [General]
I'm trying to include DAGMC in the OpenMC cmakelists.txt file with the following method
```
include(FetchContent)
FetchContent_Declare(
  dagmc
  GIT_REPOSITORY https://github.com/svalinn/DAGMC.git
  GIT_TAG v${DAGMC_VERSION}
  CMAKE_ARGS 
    -DCMAKE_INSTALL_PREFIX:PATH=<INSTALL_DIR>
    -DMOAB_DIR:PATH=${MOAB_ROOT}
    -DBUILD_STATIC_LIBS:BOOL=OFF
    -DBUILD_UWUW:BOOL=OFF
    -DBUILD_TALLY:BOOL=OFF
    -DBUILD_MAKE_WATERTIGHT:BOOL=OFF
    -DBUILD_OVERLAP_CHECK:BOOL=OFF
    -DBUILD_TESTS:BOOL=OFF
)
FetchContent_MakeAvailable(dagmc)
```

However I'm getting this error
```
      CMake Error at /tmp/tmpxthxn5gf/build/_deps/dagmc-src/cmake/DAGMC_macros.cmake:18 (message):
        Specified CMAKE_BUILD_TYPE is invalid; valid options are Release, Debug,
        RelWithDebInfo
      Call Stack (most recent call first):
        /tmp/tmpxthxn5gf/build/_deps/dagmc-src/CMakeLists.txt:54 (dagmc_setup_build)
```

which I don't understand as it [looks like](https://github.com/svalinn/DAGMC/blob/7e1cdb638b59ba22fd6ff0b09e94d8f658a1eadf/cmake/DAGMC_macros.cmake#L11-L13) the DAGMC cmakelists defaults to a release build

Has anyone had any luck including DAGMC via the FetchContent_Declare command or perhaps the ExternalProject_Add command

**Comment by gonuke** [comment]
Is `FetchContent` specifying a build type? I don't see what type is being requested.  Can you force a build type in the options you pass?

**Reply by shimwell** [reply]
I can pass in a build type and I've tried passing in the "Release" build type but this gets the same error
```
include(FetchContent)
FetchContent_Declare(
  dagmc
  GIT_REPOSITORY https://github.com/svalinn/DAGMC.git
  GIT_TAG v${DAGMC_VERSION}
  CMAKE_ARGS 
    -DCMAKE_INSTALL_PREFIX:PATH=<INSTALL_DIR>
    -DCMAKE_BUILD_TYPE=Release
    -DMOAB_DIR:PATH=${MOAB_ROOT}
    -DBUILD_STATIC_LIBS:BOOL=OFF
    -DBUILD_UWUW:BOOL=OFF
    -DBUILD_TALLY:BOOL=OFF
    -DBUILD_MAKE_WATERTIGHT:BOOL=OFF
    -DBUILD_OVERLAP_CHECK:BOOL=OFF
    -DBUILD_TESTS:BOOL=OFF
)
FetchContent_MakeAvailable(dagmc)
```

**Comment by gonuke** [comment]
It looks like we need our error message to report what build type was actually requested.

**Reply by gonuke** [reply]
`FetchContent_Declare` must be overriding your request?

**Comment by gonuke** [comment]
See PR #960 for more diagnostic help
